packages/trader/src/App/Components/Elements/VideoPlayer/video-player.tsx

```diff
@@ -32,30 +32,36 @@ const VideoPlayer = ({ src, is_mobile, data_testid }: TVideoPlayerProps) => {
     const progress_bar_ref = React.useRef<HTMLDivElement>(null);
     const progress_dot_ref = React.useRef<HTMLSpanElement>(null);
     const animation_ref = React.useRef(0);
-    const timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const play_on_rewind_timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const replay_timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
     const is_dragging = React.useRef(false);
 
-    const togglePlay = (
-        e:
-            | React.MouseEvent<HTMLDivElement | HTMLButtonElement>
-            | React.KeyboardEvent<HTMLDivElement | HTMLButtonElement>
-    ) => {
-        e.stopPropagation();
+    const togglePlay = React.useCallback(
```

Comment:
React.useCallback is needed to make sure React.memo on VideoOverlay is working properly.